### PR TITLE
Clarify dohpath manipulation attack

### LINKED ISSUE #70: Modifying dohpath
What does this mean?

> DoH resolvers that allow discovery using DNS SVCB answers over unencrypted DNS MUST NOT provide differentiated behavior based on the HTTP path alone, since an attacker could modify the "dohpath" parameter.

I'm struggling to work out what this might mean for someone operating a DoH server.

Does this mean that rewriting "/doh{?dns}" to "/~attacker/abcdef{dns}abcdef.txt" is considered something that the DoH resolver needs to identify and recover from?

Or, is this more narrow in that a DoH server cannot offer two different services on a referred endpoint with different treatment policies?

In the former case, the attacker can deny service by modifying the "dohpath".  In the latter case, the attacker gets to choose which of the two services is used.

I think that you want to lump the former under the general DoS risk.  But the latter implies something very specific in terms of action on the part of the resolver operator.  They cannot offer more than one DoH service on the same IP address (where DDR doesn't give them any virtual hosting capabilities via SNI) or hostname (where it does).  You might need to be a lot clearer about that.

### PR DESCRIPTION
**Pull request by tfpauly**
Closes #70

### CODE REVIEW

File: draft-ietf-add-ddr.md [hunk 1]
```diff
@@ -459,7 +459,13 @@ unintentionally, clients can re-send their SVCB queries periodically.
 
 DoH resolvers that allow discovery using DNS SVCB answers over unencrypted
 DNS MUST NOT provide differentiated behavior based on the HTTP path alone,
-since an attacker could modify the "dohpath" parameter.
+since an attacker could modify the "dohpath" parameter. For example, if a
+DoH resolver provides provides a filtering service for on URI path, and
```
Comment (mstojens) [inline draft-ietf-add-ddr.md:463]: for _one_ URI path (typo)

**Review by mstojens — CHANGES_REQUESTED** [review]

File: draft-ietf-add-ddr.md [hunk 2]
```diff
@@ -459,7 +459,13 @@ unintentionally, clients can re-send their SVCB queries periodically.
...
+a non-filtered service for another URI path, an attacker could select
+which of these services is used by modifying the "dohpath" parameter. Such
+differentiated services need to also rely on other aspects of connections
+and URIs being different, such as using separate resolver IP addresses and
+hostnames.
```
Comment (martinthomson) [inline draft-ietf-add-ddr.md:468]: I wonder if you need to be so definitive here.  A lot of networks are very confident in their ability to get DHCP to endpoints without these attacks.  Is it not enough to identify the risk and let people reach their own conclusions?

I mean, if it takes a backhoe and cable splicing to mount the attack and the outcome is that you get to switch a user from a filtered resolver to a unfiltered one (or vice versa), maybe that's something that doesn't require a massive service reconfiguration to counter.

So maybe you can instead say:

```suggestion
which of these services is used by modifying the "dohpath" parameter.
These attacks can be mitigated by providing separate resolver IP
addresses or hostnames.
```

**Review by martinthomson — COMMENTED** [review]
Thanks @tfpauly, this is an improvement.

File: draft-ietf-add-ddr.md (hunk 1 above)
Comment (tfpauly) [reply draft-ietf-add-ddr.md:463]:
```suggestion
DoH resolver provides provides a filtering service for one URI path, and
```

File: draft-ietf-add-ddr.md (hunk 2 above)
Comment (tfpauly) [reply draft-ietf-add-ddr.md:468]: That works for me! Simpler is good here.

**Comment by tfpauly** [comment]
@mstojens please review!!

**Review by mstojens — APPROVED** [review]
LGTM

**Pull request merged**